private/bufpkg/bufconfig/check_config.go

```diff
@@ -45,6 +45,41 @@ type CheckConfig interface {
 	// of the IDs and categories.
 	FileVersion() FileVersion
 
+	// Disabled says whether or not the given check should be entirely disabled.
+	//
+	// This happens if an ignore path matches a module directory, which is valid
+	// in cases such as:
+	//
+	//   version: v2
+	//   modules:
+	//     - path: proto
+	//     - path: vendor
+	//   lint:
+	//     ignore:
+	//       - vendor
+	//
+	// Or:
+	//
+	//   version: v2
+	//   modules:
+	//     - path: proto
+	//     - path: vendor
+	//       lint:
+	//         ignore:
+	//           - vendor
+	//
+	// We no longer produce an error in this case. Instead, we set Disabled(), and
+	// do not run checks. This means that the following is no longer an error:
+	//
+	//   version: v1
+	//   lint:
+	//     ignore:
+	//       - .
+	//
+	// We could make it so that ignore == moduleDirPath is only allowed for v2, however
```

Comment:
```suggestion
	// We could make it so that ignore == moduleDirPath is only allowed for v2, however
```
Did you mean v2 here?

Reply:
yea i updated it

Reply:
Ah ok was reviewing prior version.